File: C/Exercicies/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

// Declare the function that gets a positive float number from the user
float get_positive_float_number(void);

// Main function
int main(void)
{
    // Get a positive float number from the user and round it to the nearest cent
    float cash = get_positive_float_number();
    int cents = round(cash * 100);

    // Declare variables to keep track of the coin value and the number of coins used
    int value = 0;
    int num_coins = 0;

    // Calculate the minimum number of coins required to give the user change
    while (value < cents)
    {
        if (value + 25 <= cents)  // If a quarter can be used
        {
            value += 25;
        }
        else if (value + 10 <= cents)  // If a dime can be used
        {
            value += 10;
        }
        else if (value + 5 <= cents)  // If a nickel can be used
        {
            value += 5;
        }
        else  // If a penny can be used
        {
            value += 1;
        }
        num_coins++;
    }

    // Print the minimum number of coins required
    printf("%i\n", num_coins);
}

// Get a positive float number from user
float get_positive_float_number(void)
{
    float n;
    do
    {
        n = get_float("Money: ");
    }
    while (n <= 0);
    return n;
}

File: Python/Exercises/cash.py
# Get a positive float number from user
def get_positive_float_number():
    while True:
        try:
            # Prompt user for input
            n = float(input("Cash: "))
            # Validate input
            if n > 0:
                # Return valid input
                return n
        except ValueError:
            # Ignore input that cannot be converted to a float
            pass

def cash(money):
    coins = [25, 10, 5, 1] # Define the coin values in decreasing order
    num_coins = 0
    for coin in coins:
        while money >= coin: # While the coin can be used
            money -= coin
            num_coins += 1
    # Print the number of coins needed
    print(num_coins)

if __name__ == "__main__":
    # Get the amount of money from the user
    money = get_positive_float_number()
    # Convert the amount to cents and round to the nearest penny
    money = round(money * 100)
    # Calculate the minimum number of coins needed
    cash(money)